caffe2/python/tutorials/py_gen/MNIST.py

```diff
@@ -156,79 +212,45 @@ def AddLeNetModel(model, data):
-# 
-#     xent = model.LabelCrossEntropy([softmax, label], 'xent')
-#     
-# `AveragedLoss` will take in the cross entropy and return the average of the losses found in the cross entropy.
-# 
-#     loss = model.AveragedLoss(xent, "loss")
-# 
-# For bookkeeping purposes, we will also compute the accuracy of the model by invoking the AddAccuracy function like so:
-# 
-#     AddAccuracy(model, softmax, label)
-# 
-# The next line is the key part of the training model: we add all the gradient operators to the model. The gradient is computed with respect to the loss that we computed above.
-# 
-#     model.AddGradientOperators([loss])
-#     
-
-# The next handful of lines support a very simple stochastic gradient descent. 
-# --- TODO(jiayq): We are working on wrapping these SGD operations in a cleaner fashion, and we will update this when it is ready. For now, you can see how we basically express the SGD algorithms with basic operators. 
-# It isn't necessary to fully understand this part at the moment, but we'll walk you through the process anyway.
-# 
-# `Iter` operator is a counter for the number of iterations we run in the training. We use `brew.iter` helper function to add it to model
-# 
-#     ITER = brew.iter(model, "iter")
-# 
-# We do a simple learning rate schedule where lr = base_lr * (t ^ gamma)
-# Note that we are doing minimization, so the base_lr is negative so we are going the DOWNHILL direction. 
-# 
-#     LR = model.LearningRate(
-#         ITER, "LR", base_lr=-0.1, policy="step", stepsize=1, gamma=0.999 )
-# ONE is a constant value that is used in the gradient update. We only need to create it once, so it is explicitly placed in param_init_net.
-# 
-#     ONE = model.param_init_net.ConstantFill([], "ONE", shape=[1], value=1.0)
-# 
-# Now, for each parameter, we do the gradient updates. Note how we get the gradient of each parameter - `ModelHelper` keeps track of that. The update is a simple weighted sum: param = param + param_grad * LR
-# 
-#     for param in model.params:
-#         param_grad = model.param_to_grad[param]
-#         model.WeightedSum([param, ONE, param_grad, LR], param)        
-#         
-# We will need to checkpoint the parameters of the model periodically. This is achieved via the Checkpoint operator. It also takes in a parameter "every" so that we don't checkpoint way too often. In this case, we will say let's checkpoint every 20 iterations, which should probably be fine.
-# 
-#     model.Checkpoint([ITER] + model.params, [],
-#                    db="mnist_lenet_checkpoint_%05d.lmdb",
-#                    db_type="lmdb", every=20)
+# The next function, `AddTrainingOperators`, adds training operators to the model. Please follow inline comments to understand all the steps. We are going to use build_sgd helper function here. One can also build the whole update process him / her self. As the model object contains all the required information. Such as parameter names (`model.param`) and a mapping from 
```

Comment:
The next function, `AddTrainingOperators`, adds training operators to the model. Please follow inline comments to understand all of the steps. We are going to use the build_sgd helper function here. You can also build the whole update process yourself. The model object contains all the required information such as parameter names (`model.param`) and a mapping from ... ?